pkg/machines/scripts/install_machine.sh

```diff
@@ -89,10 +89,24 @@ fi
 # backup
 DOMAIN_FILE="`mktemp`"
 
-virsh -c "$CONNECTION_URI" -q destroy "$VM_NAME" 2>/dev/null
+virsh -c "$CONNECTION_URI" -q destroy "$VM_NAME" 2>/dev/null || true
 virsh -c "$CONNECTION_URI" -q dumpxml "$VM_NAME" > "$DOMAIN_FILE"
 virsh -c "$CONNECTION_URI" -q undefine "$VM_NAME" --managed-save
 
+handleFailure() {
+    #inject metadata, and define
+    sed "$METADATA_LINE""i $METADATA" "$DOMAIN_FILE" | virsh -c "$CONNECTION_URI" -q define /dev/stdin
```

Comment:
I don't understand this -- we land here if virt-install failed, why would that try and define the VM again?

But then again, I don't see what it did in the success case -- then virt-install should already have defined the domain?

Reply:
oh - sorry this is wrong - did not mean to move the metadata and define part.